drivers/video/fbdev/bcm2708_fb.c

```diff
@@ -883,87 +1034,149 @@ static int bcm2708_fb_probe(struct platform_device *dev)
 	if (!fw)
 		return -EPROBE_DEFER;
 
-	fb = kzalloc(sizeof(*fb), GFP_KERNEL);
-	if (!fb) {
-		ret = -ENOMEM;
-		goto free_region;
+	ret = rpi_firmware_property(fw,
+				    RPI_FIRMWARE_FRAMEBUFFER_GET_NUM_DISPLAYS,
+				    &num_displays, sizeof(u32));
+
+	/* If we fail to get the number of displays, or it returns 0, then
+	 * assume old firmware that doesn't have the mailbox call, so just
+	 * just set one display
+	 */
+	if (ret || num_displays == 0) {
+		num_displays = 1;
+		old = 1;
+		dev_err(&dev->dev,
+			"Unable to determine number of FB's. Assuming 1\n");
+		print_debug("Unable to determine number of FB's. Assuming 1\n");
+		ret = 0;
```

Comment:
Sanity check that num_displays <MAX_FRAMEBUFFERS too. Don't want buffer overruns.